gk/main.c

```diff
@@ -552,6 +552,71 @@ gk_setup_rss(struct gk_config *gk_conf)
 	return ret;
 }
 
+static void
+gk_arp_req_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused))enum lls_reply_ty ty, int *pcall_again)
+{
+	unsigned int i;
+	struct gk_lpm *ltbl = arg;
+
+	for (i = 0; i < ltbl->num_tunnels; i++) {
+		if (ltbl->tunnels[i].flow.proto == ETHER_TYPE_IPv4
+				&& memcmp(&ltbl->tunnels[i].flow.f.v4.dst,
+				map->ip_be, sizeof(struct in_addr)) == 0) {
+			ether_addr_copy(&map->ha,
+				&ltbl->tunnels[i].nexthop_mac);
```

Comment:
Don't we need some kind of concurrency control on `nexthop_mac`? Since the callback functions run on the LLS lcore and are changing `nexthop_mac` while one or more GK instances could be reading it in `encapsulate()`.